pkg/kubelet/pod_workers.go

```diff
@@ -387,18 +394,20 @@ func newPodWorkers(
 	podCache kubecontainer.Cache,
 ) PodWorkers {
 	return &podWorkers{
-		podSyncStatuses:               map[types.UID]*podSyncStatus{},
-		podUpdates:                    map[types.UID]chan podWork{},
-		lastUndeliveredWorkUpdate:     map[types.UID]podWork{},
-		terminatingStaticPodFullnames: map[string]struct{}{},
-		syncPodFn:                     syncPodFn,
-		syncTerminatingPodFn:          syncTerminatingPodFn,
-		syncTerminatedPodFn:           syncTerminatedPodFn,
-		recorder:                      recorder,
-		workQueue:                     workQueue,
-		resyncInterval:                resyncInterval,
-		backOffPeriod:                 backOffPeriod,
-		podCache:                      podCache,
+		podSyncStatuses:                map[types.UID]*podSyncStatus{},
+		podUpdates:                     map[types.UID]chan podWork{},
+		lastUndeliveredWorkUpdate:      map[types.UID]podWork{},
+		terminatingStaticPodFullnames:  map[string]struct{}{},
```

Comment:
This map should be unnecessary if managePodLoop takes ownership of preventing duplicate starts - roughly at the top of the loop:

```
var hasStarted bool
for range podUpdates {
  if !hasStarted {
    if !p.allowPodStart(pod) { // calculates whether the pod can be started
      p.workQueue.Enqueue(pod.UID, /* some resync interval */)
      klog // log that we can't start yet consistent with other log statements in this loop
      continue
    }
    hasStarted = true
  }
}
```